sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -96,13 +93,13 @@ public DiagnosticsClientContext getDiagnosticsClientContext() {
         return diagnosticsClientContext;
     }
 
-    public void recordResponse(RxDocumentServiceRequest request, StoreResult storeResult) {
+    public void recordResponse(RxDocumentServiceRequest request, StoreResult storeResult, GlobalEndpointManager globalEndpointManager) {
         Objects.requireNonNull(request, "request is required and cannot be null.");
         Instant responseTime = Instant.now();
 
         StoreResponseStatistics storeResponseStatistics = new StoreResponseStatistics();
         storeResponseStatistics.requestResponseTimeUTC = responseTime;
-        storeResponseStatistics.storeResult = storeResult;
+        storeResponseStatistics.storeResult = StoreResult.createSerializableStoreResult(storeResult);
```

Comment:
I wonder whether use a new model/class will make more sense. 

For example:
Instead of using StoreResult -> use a new class CosmosDiagnosticsStoreResult, and contain the following properties directly.
![image](https://user-images.githubusercontent.com/64233642/163652271-e88a47f1-db5d-41d6-80b8-756d6f63607c.png)

Reply:
Actually not, since we serialize all the fields from `StoreResult`, so making a copy of it would just be duplicate object of `StoreResult`. In addition to that, `StoreResult` doesn't contains any circular references, so we can use it as it is. Thoughts?
 
In my opinion, it rather makes a more compelling case for `StoreResponse` - since it contains fields that we don't need for sure, like `content`, `responseHeaderNames` and `responseHeaderValues`

But if we go into the copy path, it may take more time for this issue. 
I am interested in making those changes, however, would not want to change a lot since we want to release this as a hotfix. 

Open to making those changes in our work on re-modeling `CosmosDiagnostics` altogether.

Reply:
+1 on Kushagra's comment. To me StoreResult is basically the "model". I agree that making changes to StoreResponse and creating model for diagnostics there is preferrable.

Reply:
hmmm, I guess I was taking StoreResponseStatistics as the model itself. When we recordResponse for example, we should be able to just cherry pick the information we cared in cosmosDiagnostics (including extract information from storeResponse as well, eventually, just whatever StoreResultSerializer is doing today)